docs/design/proposals/destructive-operations/destructive-operations.md

```diff
@@ -0,0 +1,202 @@
+
+* `services.k8s.aws/`: All ACK controllers will read the policy of this namespace/resource
+* `<service>.services.k8s.aws/`: Only the <service> ACK controller will read the policy of this namespace (taking precedence over the former option)
+
+### AWS Removal Policy
+
+The name of this annotation could be either:
+
+* Follow the AWS variations (`RemovalPolicy` or `DeletionPolicy`) which would put it inline with expectations of current CDK and CloudFormation users
+* Follow existing K8s variations (`ResourcePolicy` or `ReclaimPolicy`) which would put it inline with expectations of current K8s users
+
+Users familiar with K8s would find the existing K8s variations more recognizable but may incorrectly assume that the annotation has the same values and behaves identically to native K8s resources. Therefore, it may be better to introduce the AWS terminology into the annotation - as the proposed functionality will behave identically to CDK and CloudFormation. Considering this annotation takes effect after the user calls “Delete”, I propose using `DeletionPolicy` as the annotation name.
+
+(Requirement 7) In order to account for a future possibility of adding additional policy options, the value of this annotation should be an enumeration. Considering the name already references the existing CloudFormation `DeletionPolicy` field, the values should similarly follow those values:
+
+* `Delete` - deletes the AWS resource alongside the CR
+* `Retain` - retains the AWS resource, but deletes the CR
+
+The default value for this annotation will be `Delete` as it is in-line with the existing ACK functionality of deleting the underlying AWS resource.
+
+#### Examples
+
+Definition at the controller - all resources reconciled by this controller will use the “retain” policy (unless overridden by an annotation):
+
+```yaml
+containers:
+    - command:
+    - ./bin/controller
+    args:
+    - --aws-account-id
+    - "$(AWS_ACCOUNT_ID)"
+    - --aws-region
+    - "$(AWS_REGION)"
+    - --enable-development-logging
+    - "$(ACK_ENABLE_DEVELOPMENT_LOGGING)"
+    - --log-level
+    - "$(ACK_LOG_LEVEL)"
+    - --resource-tags
+    - "$(ACK_RESOURCE_TAGS)"
++  - --deletion-policy
++  - "retain"
+```
+
+Definition at the namespace (protecting all ACK resources across any service):
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations:
+    services.k8s.aws/deletion-policy: retain
+  name: my-retained-namespace
+```
+
+Definition at the namespace (protecting only S3 ACK resources):
```

Comment:
can this made be granular upto resource at namespace level?

Ex: `bucket.s3.services.k8s.aws/deletion-policy: retain`

Reply:
That's an interesting idea. Like wildcard matching the resource type and API groups at any level.

Reply:
I was not necessarily suggesting a wildcard/regex matching, but just ability to specify that only retain a single resource from the service at the namespace level.

Although having a powerful matcher wouldn't hurt but we can just start small. :)